File: VB/E1984/SerializableFormatCollection.vb
Imports DevExpress.XtraTreeList.StyleFormatConditions
Imports System.Collections

Namespace E1984
    Public Class SerializableFormatCollection
        Inherits CollectionBase

        Public Sub New(ByVal collection As StyleFormatConditionCollection)
            CopyFromStyleFormats(collection)
        End Sub

        Public Sub New()
        End Sub

        Public Function Add() As SerializableFormatCondition
            Dim serializableFormat As SerializableFormatCondition = New SerializableFormatCondition()
            InnerList.Add(serializableFormat)
            Return serializableFormat
        End Function

        Public Sub CopyFromStyleFormats(ByVal styles As StyleFormatConditionCollection)
            Clear()

            For Each style As StyleFormatCondition In styles
                Dim serializableFormat As SerializableFormatCondition = New SerializableFormatCondition()
                serializableFormat.CopyFromStyleFormat(style)
                InnerList.Add(serializableFormat)
            Next
        End Sub

        Public Sub CopyToStyleFormats(ByVal styles As StyleFormatConditionCollection)
            styles.Clear()

            For Each serializableFormat As SerializableFormatCondition In Me
                Dim style As StyleFormatCondition = New StyleFormatCondition()
                styles.Add(style)
                serializableFormat.CopyToStyleFormat(style)
            Next
        End Sub
    End Class
End Namespace

File: CS/E1984/SerializableFormatCollection.cs
using DevExpress.XtraTreeList.StyleFormatConditions;
using System.Collections;

namespace E1984 {
    public class SerializableFormatCollection : CollectionBase
    {
        public SerializableFormatCollection(StyleFormatConditionCollection collection)
        {
            CopyFromStyleFormats(collection);
        }

        public SerializableFormatCollection()
        {
        }


        public SerializableFormatCondition Add()
        {
            SerializableFormatCondition serializableFormat = new SerializableFormatCondition();
            InnerList.Add(serializableFormat);
            return serializableFormat;
        }

        public void CopyFromStyleFormats(StyleFormatConditionCollection styles)
        {
            Clear();
            foreach (StyleFormatCondition style in styles)
            {
                SerializableFormatCondition serializableFormat = new SerializableFormatCondition();
                serializableFormat.CopyFromStyleFormat(style);
                InnerList.Add(serializableFormat);
            }
        }

        public void CopyToStyleFormats(StyleFormatConditionCollection styles)
        {
            styles.Clear();
            foreach (SerializableFormatCondition serializableFormat in this)
            {
                StyleFormatCondition style = new StyleFormatCondition();
                styles.Add(style);
                serializableFormat.CopyToStyleFormat(style);
            }
        }
    }
}
